x-pack/plugins/apm/server/lib/search_strategies/queries/query_histograms_generator.ts

```diff
@@ -9,29 +9,30 @@ import type { estypes } from '@elastic/elasticsearch';
 
 import type { ElasticsearchClient } from 'src/core/server';
 
-import type { SearchServiceFetchParams } from '../../../../../common/search_strategies/correlations/types';
+import type {
+  FieldValuePair,
+  SearchStrategyParams,
+} from '../../../../common/search_strategies/types';
 
-import type { AsyncSearchServiceLog } from '../async_search_service_log';
-import type { AsyncSearchServiceState } from '../async_search_service_state';
+import type { SearchServiceLog } from '../search_service_log';
+import type { LatencyCorrelationsSearchServiceState } from '../latency_correlations/latency_correlations_search_service_state';
 import { CORRELATION_THRESHOLD, KS_TEST_THRESHOLD } from '../constants';
 
 import { getPrioritizedFieldValuePairs } from './get_prioritized_field_value_pairs';
 import { fetchTransactionDurationCorrelation } from './query_correlation';
 import { fetchTransactionDurationRanges } from './query_ranges';
 
-import type { FieldValuePairs } from './query_field_value_pairs';
-
 export async function* fetchTransactionDurationHistograms(
```

Comment:
We have an item in the meta issue to refactor away from it: #109220 - didn't do it in this PR because it was more about code deduplication and was already getting quite large.

Reply:
Makes sense